### PR TITLE
bpo-36601: clarify signal handler comment and remove unnecessary pid check.

### PR DESCRIPTION
**Pull request by jdemeyer**
https://bugs.python.org/issue36601

### CODE REVIEW

**Comment by gpshead** [comment]
I suggest adding a NEWS entry that says something along the lines of "A believed now-meaningless internal check for getpid() == main_pid was removed from Python's signal handler."

File: Modules/signalmodule.c [hunk 1]
```diff
@@ -76,8 +76,9 @@ class sigset_t_converter(CConverter):
    When threads are supported, we want the following semantics:
```
Comment (gpshead) [inline Modules/signalmodule.c]: FWIW, CPython no longer has a concept of a non-thread-supporting BUILD.

**Review by gpshead — APPROVED** [review]

**Comment by jdemeyer** [comment]
> I suggest adding a NEWS entry that says something along the lines of "A believed now-meaningless internal check for getpid() == main_pid was removed from Python's signal handler."

Done.

**Comment by jdemeyer** [comment]
This was set to "awaiting merge" almost 2 weeks ago. Can this actually be merged please?

**Comment by merwok** [comment]
I find `believed now-meaningless internal` a little hard to parse

**Comment by gpshead** [comment]
heh thanks, i cleaned up the news entry wording.

**Review by gpshead — APPROVED** [review]
